plugins/org.locationtech.udig.tool.edit/src/org/locationtech/udig/tools/edit/impl/DeleteTool.java

```diff
@@ -112,48 +136,91 @@ public void mouseReleased( MapMouseEvent e ) {
             if (layer == null)
                 throw new Exception("No layers in map"); //$NON-NLS-1$
 
-            Envelope env = getContext().getBoundingBox(e.getPoint(), 6);
+            Envelope env = getContext().getBoundingBox(e.getPoint(), DELETE_SEARCH_SIZE);
             FeatureCollection<SimpleFeatureType, SimpleFeature>  results = getContext().getFeaturesInBbox(
                     layer,
                     env);
 
             reader = results.features();
-            
+
             final boolean found=!reader.hasNext() ;
-        	getContext().updateUI(new Runnable() {
-				public void run() {
+            getContext().updateUI(new Runnable() {
+                public void run() {
                     if( getContext().getActionBars()==null )
                         return;
                     IStatusLineManager bar = getContext().getActionBars().getStatusLineManager();
-	            	if (bar!=null){
-	            		if( found )
-	            			bar.setErrorMessage( Messages.DeleteTool_warning);
-	            		else
-	            			bar.setErrorMessage(null);
-	            	}
-				}
-			});
+                    if (bar!=null){
+                        if( found )
+                            bar.setErrorMessage( Messages.DeleteTool_warning);
+                        else
+                            bar.setErrorMessage(null);
+                    }
+                }
+            });
 
             if (found)
                 return;
 
-            SimpleFeature feature=reader.next();
+            final SimpleFeature[] features = results.toArray(new SimpleFeature[]{});
+            if (features.length == 1) {
+                SimpleFeature feature=features[0];
+                if (DELETE_CONFIRM && !MessageDialog.openConfirm(null, "", MessageFormat.format(
+                        Messages.DeleteTool_confirmation_text2, feature.getIdentifier()))) {
+                    return;
+                }
+                MapCommand deleteFeatureCommand = getContext().getEditFactory().createDeleteFeature(feature, layer);
+                getContext().sendASyncCommand(deleteFeatureCommand);
+            } else {
+                final Menu menu = new Menu(((ViewportPane) e.source).getControl().getShell(), SWT.POP_UP);
+                final ILayer selectedLayer = layer;
+                Display.getDefault().asyncExec(new Runnable() {
+
+                    @Override
+                    public void run() {
+                    	final String attribName = FeatureUtils.getActualPropertyName(
```

Comment:
OK I modify FeatureUtils.getActualPropertyName(..) to perform first a strict case sensitive checking for attribute name before resorting to a case insensitive one.